src/iosource/ManagerBase.cc

```diff
@@ -129,13 +136,17 @@ void ManagerBase::FindReadySources(std::vector<IOSource*>* ready)
 					return;
 					}
 				}
+
+			if ( ! time_to_poll && src->src->IsPacketSource() && ((PktSrc*)src->src)->IsLive() )
```

Comment:
Wouldn't we want to fail out of this as fast as possible? We're not always going to do the cast if it's not a packet source (which is most of the sources). Calling dynamic_cast on every source adds a bit of extra overhead here that's not really necessary.

Reply:
Never mind, I decided that it's probably dwarfed by lots of other things in that area of the code and changed it to a `dynamic_cast`.

Reply:
You make a good point though about considering slower performance of `dynamic_cast` versus the virtual function call, but yeah if in the end you measure it and can't really see any difference since it's dwarfed by other things, it maybe slightly better to go with it than add API clutter.